server/client.go

```diff
@@ -2123,7 +2123,10 @@ func (c *client) processSub(argo []byte, noForward bool) (*subscription, error)
 	default:
 		return nil, fmt.Errorf("processSub Parse Error: '%s'", arg)
 	}
+	return c.processSub(sub, noForward)
+}
 
+func (c *client) processSub(sub *subscription, noForward bool) (*subscription, error) {
```

Comment:
@derekcollison It seems that on Sublist.Insert() error, we send an error to the client but return `nil, nil` (no subscription, no error), which forces the callers to possibly check for nil sub. I wonder if I should not change that to return `nil, err` but maybe in parseSub() to restore old behavior (that is no error returned otherwise parser will close connection)?

Reply:
I have a second commit with this change and I think it is better.

Reply:
Unclear, I know we did not want to return an error for some reason.